tools/shared/src/main/scala/org/scalajs/core/tools/sem/Semantics.scala

```diff
@@ -16,7 +16,9 @@ import org.scalajs.core.tools.optimizer.LinkedClass
 final class Semantics private (
     val asInstanceOfs: CheckedBehavior,
     val strictFloats: Boolean,
-    val runtimeClassName: Semantics.RuntimeClassNameFunction) {
+    val runtimeClassName: Semantics.RuntimeClassNameFunction,
+    val moduleInit: CheckedBehavior
```

Comment:
I'd like to keep the CheckedBehaviors together, before `strictFloats`. The rest of the class should also follow that ordering.